src/NuGet.Clients/NuGet.CommandLine/RegistryKeyUtility.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Security;
+using Microsoft.Win32;
+using NuGet.Common;
+
+namespace NuGet.CommandLine
+{
+    public static class RegistryKeyUtility
+    {
+        public static object GetValueFromRegistryKey(string name, string registryKeyPath, RegistryKey registryKey, ILogger logger)
+        {
+            try
+            {
+                var key = registryKey?.OpenSubKey(name);
+
+                var result = key?.GetValue(name);
+
+                if (key != null)
+                {
+                    // Note that according to MSDN, this method does nothing if you call it on an instance of RegistryKey that is already closed.
+                    key.Close();
```

Comment:
shouldn't this `key.Close()` be replicated in catch block as well? so that if key is not closed when exception is thrown, it can be closed in catch block.